File: scratch/check_braces.ts
import fs from 'fs';

const content = fs.readFileSync('/Users/patrickhuber/Documents/Antigravity Projects/Groovelab app/src/components/TeacherDashboard.tsx', 'utf8');

let openBraces = 0;
let closeBraces = 0;
let openParens = 0;
let closeParens = 0;

for (let i = 0; i < content.length; i++) {
    if (content[i] === '{') openBraces++;
    if (content[i] === '}') closeBraces++;
    if (content[i] === '(') openParens++;
    if (content[i] === ')') closeParens++;
}

console.log(`Braces: Open=${openBraces}, Close=${closeBraces}, Balance=${openBraces - closeBraces}`);
console.log(`Parens: Open=${openParens}, Close=${closeParens}, Balance=${openParens - closeParens}`);

File: scratch/check_braces.py
import re

with open('/Users/patrickhuber/Documents/Antigravity Projects/Groovelab app/apps/groovelab/src/components/StudentAvatarDashboard.tsx', 'r') as f:
    lines = f.readlines()

# Let's inspect specific lines around the errors
error_lines = [5850, 8541, 9106, 9108, 11091, 12412, 13554]

for el in error_lines:
    print(f"=== LINE {el} ===")
    start = max(0, el - 10)
    end = min(len(lines), el + 10)
    for idx in range(start, end):
        print(f"{idx+1}: {lines[idx]}", end="")
